src/components/vice/admin/table/actionButtons.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from "react";
+import { useQuery } from "react-query";
+
+import { Button, Popper } from "@material-ui/core";
+import { Skeleton } from "@material-ui/lab";
+
+import { getMessage as msg } from "@cyverse-de/ui-lib";
+
+import { id } from "./functions";
+import useStyles from "./styles";
+
+import { asyncData } from "../../../../serviceFacades/vice/admin";
+
+const ActionButtonsSkeleton = () => {
+    return (
+        <Skeleton variant="rect" animation="wave" height={75} width="100%" />
+    );
+};
+
+const ActionButton = ({ baseID, name, handler, onClick, popperMsgKey }) => {
+    const classes = useStyles();
+    return (
+        <Button
+            id={id(baseID, "button", name)}
+            variant="contained"
+            color="primary"
+            onClick={(event) => {
+                event.preventDefault();
+                event.stopPropagation();
+                return onClick(event, handler, popperMsgKey);
+            }}
+            className={classes.actionButton}
+        >
+            {msg(name)}
+        </Button>
+    );
+};
+
+export default ({
+    row,
+    handleExtendTimeLimit = (_) => {},
+    handleDownloadInputs = (_) => {},
+    handleUploadOutputs = (_) => {},
+    handleExit = (_) => {},
+    handleSaveAndExit = (_) => {},
+}) => {
+    const classes = useStyles();
+
+    const [anchorEl, setAnchorEl] = useState(null);
+    const [open, setOpen] = useState(false);
+    const [popperMessage, setPopperMessage] = useState("");
+
+    const { status, data, error } = useQuery(
+        ["async-data", row.externalID],
+        asyncData
+    );
+
+    const isLoading = status === "loading";
+    const hasErrored = status === "error";
+
+    if (hasErrored) {
+        console.log(error);
+    }
+
+    const onClick = (event, dataFn, msgKey) => {
+        let tlErr;
+        let tlData;
+
+        try {
+            tlData = dataFn(data.analysisID, row.externalID);
+        } catch (err) {
+            tlErr = err;
+        }
+
+        setAnchorEl(event.currentTarget);
+        setPopperMessage(tlErr ? tlErr.message : msg(msgKey));
+        setOpen(true);
+
+        return tlData;
+    };
```

Comment:
Instead of the using the `Popper` component, we can use the `annoucer` that shows a `toast` message.
`import {
    announce,
    AnnouncerConstants,
} from "@cyverse-de/ui-lib";`
`announce({
                text: <errorMessage>,
                variant: AnnouncerConstants.ERROR,
 });`

Reply:
Alrighty, this should be switched over to an Announcer. The switch over also simplified the code a good bit and exposed some i18n issues lurking about, so thanks for the suggestion.